From 2f080b05736e93135ae2fc4eb79b97baaaf9bc35 Mon Sep 17 00:00:00 2001
Message-Id: <2f080b05736e93135ae2fc4eb79b97baaaf9bc35.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Wed, 9 Feb 2011 10:43:27 +0800
Subject: [PATCH 002/312] REVERTME [VPG]: drm: disable GPU authentication

Conflicts:

	drivers/gpu/drm/drm_fops.c
(cherry picked from commit 061a3576e15c03b890ec4f91b2b7787e4f73f050)

Change-Id: I13118e41d2b5c21a78cdb0958aeac8df4bc62a29
Signed-off-by: Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/Kconfig    |    7 +++++++
 drivers/gpu/drm/drm_fops.c |    5 +++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/Kconfig b/drivers/gpu/drm/Kconfig
index f512004..66f6bca 100644
--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -30,6 +30,13 @@ config DRM_USB
 	depends on USB_SUPPORT && USB_ARCH_HAS_HCD
 	select USB
 
+config NO_GPU_AUTHENTICATION
+	bool "Disable GPU authentication"
+	depends on DRM
+	default n
+	help
+	  Choose this option if you want to disable GPU authentication.
+
 config DRM_KMS_HELPER
 	tristate
 	depends on DRM
diff --git a/drivers/gpu/drm/drm_fops.c b/drivers/gpu/drm/drm_fops.c
index 021fe5d..2be326f 100644
--- a/drivers/gpu/drm/drm_fops.c
+++ b/drivers/gpu/drm/drm_fops.c
@@ -203,8 +203,13 @@ static int drm_open_helper(struct file *filp, struct drm_minor *minor)
 	priv->minor = minor;
 
 	/* for compatibility root is always authenticated */
+#ifdef CONFIG_NO_GPU_AUTHENTICATION
+	priv->always_authenticated = 1;
+	priv->authenticated = 1;
+#else
 	priv->always_authenticated = capable(CAP_SYS_ADMIN);
 	priv->authenticated = priv->always_authenticated;
+#endif
 	priv->lock_count = 0;
 
 	INIT_LIST_HEAD(&priv->lhead);
-- 
1.7.9.5

